internal/mode/static/handler.go

```diff
@@ -164,11 +172,49 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, logger logr.Log
 		h.cfg.graphBuiltHealthChecker.setAsReady()
 	}
 
-	var err error
+	// TODO(sberman): hardcode this deployment name until we support provisioning data planes
+	// If no deployments exist, we should just return without doing anything.
+	deploymentName := types.NamespacedName{
+		Name:      "tmp-nginx-deployment",
+		Namespace: h.cfg.gatewayPodConfig.Namespace,
+	}
+
+	// TODO(sberman): if nginx Deployment is scaled down, we should remove the pod from the ConnectionsTracker
+	// and Deployment.
+	// If fully deleted, then delete the deployment from the Store and close the stopCh.
+	stopCh := make(chan struct{})
+	deployment := h.cfg.nginxDeployments.GetOrStore(deploymentName, stopCh)
+	if deployment == nil {
+		panic("expected deployment, got nil")
```

Comment:
Should we close the stopCh here?

Reply:
Is it worth it when a panic occurs?

Reply:
meh, probably not. However, we should probably close the channel on a graceful shutdown... can we leverage the context here?

Reply:
I could also have the broadcaster return when the context is done. The channel will get garbage collected.